tests/lisa/test_energy_model.py

```diff
@@ -0,0 +1,253 @@
+    (1500, ActiveState(capacity=150, power=150)),
+    (2000, ActiveState(capacity=200, power=200)),
+])
+
+little_cpu_idle_states = OrderedDict([
+    ('WFI',              5),
+    ('cpu-sleep-0',      0),
+    ('cluster-sleep-0',  0),
+])
+
+littles=[0, 1]
+def little_cpu_node(cpu):
+    return EnergyModelNode(cpu=cpu,
+                           active_states=little_cpu_active_states,
+                           idle_states=little_cpu_idle_states)
+
+big_cluster_active_states = OrderedDict([
+    (3000, ActiveState(power=30)),
+    (4000, ActiveState(power=40)),
+])
+
+big_cluster_idle_states = OrderedDict([
+    ('WFI',              8),
+    ('cpu-sleep-0',      8),
+    ('cluster-sleep-0',  2),
+])
+
+big_cpu_active_states = OrderedDict([
+    (3000, ActiveState(capacity=300, power=300)),
+    (4000, ActiveState(capacity=400, power=400)),
+])
+
+big_cpu_idle_states = OrderedDict([
+    ('WFI',              9),
+    ('cpu-sleep-0',      0),
+    ('cluster-sleep-0',  0),
+])
+
+bigs=[2, 3]
+
+def big_cpu_node(cpu):
+    return EnergyModelNode(cpu=cpu,
+                           active_states=big_cpu_active_states,
+                           idle_states=big_cpu_idle_states)
+
+em = EnergyModel(
+    root_node=EnergyModelRoot(children=[
+        EnergyModelNode(name='cluster_little',
+                        active_states=little_cluster_active_states,
+                        idle_states=little_cluster_idle_states,
+                        children=[little_cpu_node(0),
+                                  little_cpu_node(1)]),
+        EnergyModelNode(name='cluster_big',
+                        active_states=big_cluster_active_states,
+                        idle_states=big_cluster_idle_states,
+                        children=[big_cpu_node(2),
+                                  big_cpu_node(3)])
+    ]),
+    root_power_domain=PowerDomain(idle_states=[], children=[
+        PowerDomain(
+            idle_states=['cluster-sleep-0'],
+            children=[PowerDomain(idle_states=['WFI', 'cpu-sleep-0'], cpu=c)
+                      for c in littles]),
+        PowerDomain(
+            idle_states=['cluster-sleep-0'],
+            children=[PowerDomain(idle_states=['WFI', 'cpu-sleep-0'], cpu=c)
+                      for c in bigs]),
+        ]),
+    freq_domains=[littles, bigs]
+)
+
+class TestOptimalPlacement(TestCase):
+    def assertPlacementListEqual(self, l1, l2):
+        """
+        Assert that a pair of lists of lists contain the same lists in any order
+        """
+        s1 = set([tuple(l) for l in l1])
+        s2 = set([tuple(l) for l in l2])
+        self.assertSetEqual(s1, s2)
+
+    def test_single_small(self):
+        placements = em.get_optimal_placements({'task0': 1})
+        self.assertPlacementListEqual(placements, [[1, 0, 0, 0],
+                                                   [0, 1, 0, 0]])
+
+    def test_single_big(self):
+        placements = em.get_optimal_placements({'task0': 350})
```

Comment:
Yeah, this was a deliberate decision at the beginning of writing it, because I thought if I expressed it in terms of the data I'd probably make the exact same mistakes as I'd made in the code under test. So I wanted to work them out by hand. 

However I agree that as the code grew it didn't start to look like such a smart idea. I'm going to keep it as it is for now, but add a comment. Probably if we need to extend these tests it will be faster to add new ones in terms of the data rather than go back and re-factor the hard coded stuff.